app/team.go

```diff
@@ -1267,14 +1302,12 @@ func (a *App) InviteNewUsersToTeam(emailList []string, teamId, senderId string)
 
 	if len(invalidEmailList) > 0 {
 		s := strings.Join(invalidEmailList, ", ")
-		err := model.NewAppError("InviteNewUsersToTeam", "api.team.invite_members.invalid_email.app_error", map[string]interface{}{"Addresses": s}, "", http.StatusBadRequest)
-		return err
+		return model.NewAppError("InviteNewUsersToTeam", "api.team.invite_members.invalid_email.app_error", map[string]interface{}{"Addresses": s}, "", http.StatusBadRequest)
 	}
 
 	nameFormat := *a.Config().TeamSettings.TeammateNameDisplay
-	a.Srv().EmailService.SendInviteEmails(team, user.GetDisplayName(nameFormat), user.Id, emailList, a.GetSiteURL())
+	return a.Srv().EmailService.SendInviteEmails(team, user.GetDisplayName(nameFormat), user.Id, emailList, a.GetSiteURL())
```

Comment:
One thing that we should be careful with is that in the past we didn't return errors and now we're doing it so maybe that would end-up failing when right now we're not failing. Are we sure that returning the error is not going to provoke problems in the public API?

Reply:
good question, but I can only guess!
doing a search in the [webapp repo](https://github.com/mattermost/mattermost-webapp/search?q=sendEmailInvitesToTeamGracefully&unscoped_q=sendEmailInvitesToTeamGracefully) only returns two results. one is within a `try` block and the other is a test. So initially I'd say it wont, but there might be plugins using that feature that might fail or not expect such response.
i will submit a PR for the api docs and add a note in the PR about the breaking changes, but I'm not aware of any other means of comunicating/checking it.